ecs-java-s3-workshop/src/main/java/com/emc/ecs/s3/sample/ECSS3Factory.java

```diff
@@ -50,19 +50,40 @@
     /* a unique object name to store */
     public static final String S3_OBJECT = "workshop-object";
 
-    public static S3Client getS3Client() throws URISyntaxException {
-        // for client-side load balancing
-        //S3Config config = new S3Config(Protocol.HTTPS, S3_HOST1, S3_HOST2);
-        // ditto with multiple VDCs
-        //S3Config config = new S3Config(Protocol.HTTPS, new Vdc(S3_V1_HOST), new Vdc(S3_V2_HOST));
-
-        S3Config config = new S3Config(new URI(S3_URI));
+    public static S3Client getS3Client() {
+        return S3ClientSingleton.getInstance();
+    }
 
-        config.withIdentity(S3_ACCESS_KEY_ID).withSecretKey(S3_SECRET_KEY);
+    private static class S3ClientSingleton {
```

Comment:
You might refer to a better example of singleton in https://refactoring.guru/design-patterns/singleton/java/example#example-2, you could achieve thread-safe.